lib/identity_cache/configuration_dsl.rb

```diff
@@ -49,6 +49,9 @@ def cache_index(*fields, unique: false)
             def fetch_by_#{field_list}(#{arg_list}, includes: nil)
               id = fetch_id_by_#{field_list}(#{arg_list})
               id && fetch_by_id(id, includes: includes)
+
+            rescue RangeError
+              raise ActiveRecord::RecordNotFound
```

Comment:
This method shouldn't raise.  The exception throwing variant is below and already does a `raise ActiveRecord::RecordNotFound` if this method returns `nil`.

If we want to workaround this issue in identity cache, then it should probably be done in fetch_by_id and dynamic_attribute_cache_miss for `unique_index`.  That way the fix would at least be more consistently applied across identity cache.